.github/workflows/ci.yaml

```diff
@@ -0,0 +1,107 @@
+on: [push, pull_request]
+
+name: Continuous integration
+
+jobs:
+  check:
+    name: Check and Lint
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        rust: [stable]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - uses: actions/cache@v2
+        # https://github.com/actions/cache/blob/main/examples.md#rust---cargo
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-check-${{ hashFiles('**/Cargo.lock') }}
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          profile: minimal
+          override: true
+          components: rustfmt, clippy
+      - name: Cargo check
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --workspace --all-targets
+      - name: Cargo fmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
+      - name: Cargo clippy
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --workspace --all-targets -- -D warnings
+  test:
+    name: Run unit tests
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        rust: [stable]
+        tests: ["", "--release", "--features=static"]
```

Comment:
I think you probably want to test with libsodium also. The default encryption lib has terrible performance and the devs have said it shoudn't be used for more than testing. Can't remember where I saw that exactly but I tested it and performance was indeed terrible. So as soon as someone want's to use ZMQ in a public environment, it's gonna be needed. Whether I would trust ZMQ in a public setting is another issue however.

Reply:
Thanks I'll add it to the test matrix. 

You'll see some commits fly by while I try to get mingw to work as well but it's a bit of a nightmare it would seem ;-)

Any platforms you care about that should be covered?

Reply:
If tweetnacl isn't supposed to be used - maybe we should just make libsodium the default instead?